### PR TITLE
Centrifuge update to address mkfifo clashes

### LINKED ISSUE #422: Centrifuge error : (ERR): mkfifo(/tmp/72.inpipe1) failed.
### Description of the bug

I downloaded the pre-built centrifuge database from https://benlangmead.github.io/aws-indexes/k2 (the full link is https://genome-idx.s3.amazonaws.com/centrifuge/nt_2018_3_3.tar.gz) to use with taxprofiler. It does not run as expected. I first brought this up [in slack](https://nfcore.slack.com/archives/C031QH57DSS/p1701975837271399). 

This is my database file (the listed kraken2 databases worked fine):
```
tool,db_name,db_params,db_path
kraken2,nt,,../data/k2_nt_20230502.tar.gz
kraken2,viral,,../data/k2_viral_20231009.tar.gz
centrifuge,nt_2018,,../data/nt_2018_3_3.tar.gz
```

This is the error message:
```
Workflow execution completed unsuccessfully
The exit status of the task that caused the workflow execution to fail was: 17

Error executing process > 'NFCORE_TAXPROFILER:TAXPROFILER:PROFILING:CENTRIFUGE_CENTRIFUGE (nt_2018|B002f_pe)'

Caused by:
  Process `NFCORE_TAXPROFILER:TAXPROFILER:PROFILING:CENTRIFUGE_CENTRIFUGE (nt_2018|B002f_pe)` terminated with an error exit status (17)

Command executed:

  ## we add "-no-name ._" to ensure silly Mac OSX metafiles files aren't included
  db_name=`find -L nt_2018_3_3 -name "*.1.cf" -not -name "._*"  | sed 's/\.1.cf$//'`
  centrifuge \
      -x $db_name \
      -p 12 \
      -1 B002f_VH00770.unmapped_1.fastq.gz -2 B002f_VH00770.unmapped_2.fastq.gz \
      --report-file B002f_pe_VH00770_nt_2018.centrifuge.report.txt \
      -S B002f_pe_VH00770_nt_2018.centrifuge.results.txt \
       \
       \
  
  
  cat <<-END_VERSIONS > versions.yml
  "NFCORE_TAXPROFILER:TAXPROFILER:PROFILING:CENTRIFUGE_CENTRIFUGE":
      centrifuge: $( centrifuge --version  | sed -n 1p | sed 's/^.*centrifuge-class version //')
  END_VERSIONS

Command exit status:
  17

Command output:
  (empty)

Command error:
  INFO:    Environment variable SINGULARITYENV_NXF_TASK_WORKDIR is set, but APPTAINERENV_NXF_TASK_WORKDIR is preferred
  INFO:    Environment variable SINGULARITYENV_NXF_DEBUG is set, but APPTAINERENV_NXF_DEBUG is preferred
  (ERR): mkfifo(/tmp/72.inpipe1) failed.
  Exiting now ...

Work dir:
  /Volumes/IDGenomics_NAS/Bioinformatics/eriny/mosquito/2023-12-01/work/11/33708dd319d5e790a7d5b44e106865

Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line
```

### Command used and terminal output

```console
nextflow run nf-core/taxprofiler \
		 -profile singularity \
		 --input sample_sheet.csv \
		 --databases database.txt \
		 --outdir taxprofiler \
		 --run_kraken2 \
		 --run_centrifuge \
		 -with-tower \
		 -resume \
		 --hostremoval_reference reference.fasta \
		 --save_hostremoval_bam true \
		 --perform_shortread_hostremoval \
		 --save_hostremoval_unmapped true \
		 --perform_shortread_qc \
		 -c time.config
```

```
N E X T F L O W  ~  version 23.10.0
Launching `https://github.com/nf-core/taxprofiler` [desperate_cuvier] DSL2 - revision: 3d4eda2dbb [master]


------------------------------------------------------
                                        ,--./,-.
        ___     __   __   __   ___     /,-._.--~'
  |\ | |__  __ /  ` /  \ |__) |__         }  {
  | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                        `._,._,'
  nf-core/taxprofiler v1.1.2-g3d4eda2
------------------------------------------------------
Core Nextflow options
  revision                     : master
  runName                      : desperate_cuvier
  containerEngine              : singularity
  launchDir                    : /Volumes/IDGenomics_NAS/Bioinformatics/eriny/mosquito/2023-12-01
  workDir                      : /Volumes/IDGenomics_NAS/Bioinformatics/eriny/mosquito/2023-12-01/work
  projectDir                   : /home/eriny/.nextflow/assets/nf-core/taxprofiler
  userName                     : eriny
  profile                      : singularity
  configFiles                  : 

Input/output options
  input                        : sample_sheet.csv
  databases                    : database_all.txt
  outdir                       : taxprofiler

Preprocessing short-read QC options
  perform_shortread_qc         : true

Preprocessing host removal options
  perform_shortread_hostremoval: true
  hostremoval_reference        : reference.fasta
  save_hostremoval_bam         : true
  save_hostremoval_unmapped    : true

Profiling options
  run_centrifuge               : true
  run_kraken2                  : true
  ganon_report_rank            : default

!! Only displaying parameters that differ from the pipeline defaults !!
------------------------------------------------------
If you use nf-core/taxprofiler for your analysis please cite:

* The pipeline
  https://doi.org/10.5281/zenodo.7728364

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://github.com/nf-core/taxprofiler/blob/master/CITATIONS.md

Monitor the execution with Nextflow Tower using this URL: https://tower.nf/user/erin-olde/watch/5y7UeijszzHmnN
executor >  local (4)
[11/307bce] process > NFCORE_TAXPROFILER:TAXPROFILER:INPUT_CHECK:SAMPLESHEET_CHECK (sample_sheet.csv)             [100%] 1 of 1, cached: 1 ✔
[8e/78ff0d] process > NFCORE_TAXPROFILER:TAXPROFILER:DB_CHECK:UNTAR (k2_nt_20230502.tar.gz)                       [100%] 3 of 3, cached: 3 ✔
[d7/b16a2d] process > NFCORE_TAXPROFILER:TAXPROFILER:FASTQC (B016g)                                               [100%] 16 of 16, cached: 16 ✔
[-        ] process > NFCORE_TAXPROFILER:TAXPROFILER:SHORTREAD_PREPROCESSING:SHORTREAD_FASTP:FASTP_SINGLE         -
[c8/a8b135] process > NFCORE_TAXPROFILER:TAXPROFILER:SHORTREAD_PREPROCESSING:SHORTREAD_FASTP:FASTP_PAIRED (B013f) [100%] 16 of 16, cached: 16 ✔
[41/94fab5] process > NFCORE_TAXPROFILER:TAXPROFILER:SHORTREAD_PREPROCESSING:FASTQC_PROCESSED (B013g)             [100%] 16 of 16, cached: 16 ✔
[97/b9c3b6] process > NFCORE_TAXPROFILER:TAXPROFILER:SHORTREAD_HOSTREMOVAL:BOWTIE2_BUILD (reference.fasta)        [100%] 1 of 1, cached: 1 ✔
[47/54c4f2] process > NFCORE_TAXPROFILER:TAXPROFILER:SHORTREAD_HOSTREMOVAL:BOWTIE2_ALIGN (B015g)                  [100%] 16 of 16, cached: 16 ✔
[6d/fc97f5] process > NFCORE_TAXPROFILER:TAXPROFILER:SHORTREAD_HOSTREMOVAL:SAMTOOLS_INDEX (B016f)                 [100%] 16 of 16, cached: 16 ✔
[98/78ecba] process > NFCORE_TAXPROFILER:TAXPROFILER:SHORTREAD_HOSTREMOVAL:SAMTOOLS_STATS (B021f)                 [ 93%] 15 of 16, cached: 15
[b0/29c8c0] process > NFCORE_TAXPROFILER:TAXPROFILER:PROFILING:KRAKEN2_KRAKEN2 (nt|kraken2|B023f_pe)              [ 15%] 5 of 32, cached: 5
[60/1788d7] process > NFCORE_TAXPROFILER:TAXPROFILER:PROFILING:CENTRIFUGE_CENTRIFUGE (nt_2018|B013f_pe)           [ 12%] 2 of 16, cached: 2
[-        ] process > NFCORE_TAXPROFILER:TAXPROFILER:PROFILING:CENTRIFUGE_KREPORT                                 [  0%] 0 of 2
[-        ] process > NFCORE_TAXPROFILER:TAXPROFILER:CUSTOM_DUMPSOFTWAREVERSIONS                                  -
[-        ] process > NFCORE_TAXPROFILER:TAXPROFILER:MULTIQC                                                      -

```

### Relevant files

_No response_

### System information

Nextflow version : 23.10.0 build 5889
Hardware : Local
Executor : local 
Container engine: Singularity
OS : CentOS
Version of nf-core/taxprofiler : current

### PR DESCRIPTION
**Pull request by jfy133**
Hopefully :crossed_fingers: fixes #422 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,12 +8,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### `Added`
 
 - [#439](https://github.com/nf-core/taxprofiler/pull/439) Read deduplication with fastp (added by @maxibor)
-- [#440](https://github.com/nf-core/taxprofiler/pull/440) Include pre-built kaiju databases in tutorial.md (added by @Joon-Klaps)
+- [#440](https://github.com/nf-core/taxprofiler/pull/440) Include mention of pre-built kaiju databases in tutorial.md (added by @Joon-Klaps)
+- [#424](https://github.com/nf-core/taxprofiler/pull/424) Updated to nf-core pipeline template v2.12 (added by @sofstam)
 
 ### `Fixed`
 
+- [#444](https://github.com/nf-core/taxprofiler/pull/440) Centrifuge uses dedicated /tmp directory to hopefully prevent mkfifo clashes (❤️ to @erinyoung for reporting, fix by @jfy133)
```
Comment (jfy133) [inline CHANGELOG.md:16]:
```suggestion
- [#444](https://github.com/nf-core/taxprofiler/pull/440) Centrifuge now uses dedicated tmp directory to hopefully prevent mkfifo clashes (❤️ to @erinyoung for reporting, fix by @jfy133)
```

**Comment by LilyAnderssonLee** [comment]
Super!!! Well done!!! I will test it now and will approve it once the job is completed.

**Comment by jfy133** [comment]
I hope it works :crossed_fingers:

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,12 +8,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### `Added`
 
 - [#439](https://github.com/nf-core/taxprofiler/pull/439) Read deduplication with fastp (added by @maxibor)
-- [#440](https://github.com/nf-core/taxprofiler/pull/440) Include pre-built kaiju databases in tutorial.md (added by @Joon-Klaps)
+- [#440](https://github.com/nf-core/taxprofiler/pull/440) Include mention of pre-built kaiju databases in tutorial.md (added by @Joon-Klaps)
+- [#424](https://github.com/nf-core/taxprofiler/pull/424) Updated to nf-core pipeline template v2.12 (added by @sofstam)
 
 ### `Fixed`
 
+- [#444](https://github.com/nf-core/taxprofiler/pull/440) Centrifuge now uses dedicated tmp directory to hopefully prevent mkfifo clashes (❤️ to @erinyoung for reporting, fix by @jfy133)
```
Comment (LilyAnderssonLee) [inline CHANGELOG.md:16]:
```suggestion
- [#444](https://github.com/nf-core/taxprofiler/pull/444) Centrifuge now uses dedicated tmp directory to hopefully prevent mkfifo clashes (❤️ to @erinyoung for reporting, fix by @jfy133)
```

**Review by LilyAnderssonLee — APPROVED** [review]
The test run on my data worked perfectly. Thanks for resolving this issue!

**Comment by sofstam** [comment]
Do we need to update taxpasta?

**Comment by jfy133** [comment]
@sofstam I can't see your comment now, but yes I'm a bit concerned that the tests are failing :( need to investigate what that's meaning...

**Comment by sofstam** [comment]
> @sofstam I can't see your comment now, but yes I'm a bit concerned that the tests are failing :( need to investigate what that's meaning...

I deleted it, I think Lili has an updated comment now.

**Comment by LilyAnderssonLee** [comment]
@jfy133  Have you tested the new version of `Centrifuge-kreport` in standalone mode?

**Comment by jfy133** [comment]
No I didn't... my understanding from the release it wasn't meant to change...

https://github.com/DaehwanKimLab/centrifuge/releases/tag/v1.0.4.1

How does it look in your own test run/data?

**Comment by jfy133** [comment]
OK found the problem, it was a bug in my bioconda recipe (bad docs)... going to go back and update the recipe and the biocontainer and the moduel... :sweat_smile:

**Review by LilyAnderssonLee — APPROVED** [review]
The test run for my real data was finished successfully this time. Got all correct files!!!
LGTM!

**Pull request merged**